flow.py

```diff
@@ -57,17 +62,70 @@ def dest(self):
     def dest(self, dest):
         raise AttributError("Cannot modify a flow's destination node.")
 
-    def make_packet(self, id, src, dest, data):
+    def check_last(self):
+        """Method that checks if all packets have been acknowledged and
+        updating the NetworkSimulator object accordingly
+        """
+        if self.last_acknowledged == self.num_packets:
+            self.ns.num_active_flows -= 1
```

Comment:
i believe there is now a method for this instead of directly changing the variable